zerver/lib/cache.py

```diff
@@ -278,7 +278,9 @@ def generic_bulk_cached_fetch(
         cached_objects[key] = extractor(cached_objects_compressed[key][0])
     needed_ids = [object_id for object_id in object_ids if
                   cache_keys[object_id] not in cached_objects]
-    db_objects = query_function(needed_ids)
+
+    # Only call query_function if there are some ids to fetch from the database:
+    db_objects = query_function(needed_ids) if needed_ids else []
```

Comment:
I would prefer to do this with a proper if/else block so that coverage will reveal which path was followed.

Also, I think we should have a similar check at the very top to avoid doing memcached queries on an empty set of objects.

Reply:
Did both things, and added a test for the case at the top.